### DISCUSSION TITLE
Manage port positions on a cell

### DISCUSSION THREAD

**Discussion by seekeramento** [clientIO/joint#1781] [Q&A]
I can use 
attrs: {
    portBody: { cy: - 220 }    // I assume here that my selector is a circle
  }

However, this becomes too difficult and messy to manage when dealing with multiple ports on the same cell with the assumption that we have ports at different positions, top, bottom, left, and right. 

What is the recommended way to set  a position directly for a port instead of calculating its relative position to the parent cell?
Are there examples that deal with multiple ports? 
Is there a better way to automatically have many ports to manage starting new sources at different locations of a source instead of having predefined ports? At lest if it is not possible, need to find an easier way to manage the layout of these ports.

**Comment by jamesgeorgewilliams** [comment]
In JointJS, you can create port definitions for a single port or a group of ports. These definitions can be used to manage port positions. 

In the following port definition, `position` is used to set the port layout to the `left`:

```js
const portsIn = {
        position: {
            name: 'left'
        },
        attrs: {
            portBody: {
                magnet: true,
                r: 10,
                fill: '#023047',
                stroke: '#023047'
            }
        },
        label: {
            position: {
                name: 'left',
                args: { y: 6 } 
            },
            markup: [{
                tagName: 'text',
                selector: 'label',
                className: 'label-text'
            }]
        },
        markup: [{
            tagName: 'circle',
            selector: 'portBody'
        }]
};
```

You can read more about how to create port definitions in the following tutorial:
https://resources.jointjs.com/tutorial/ports

We also have a new ports demo where you can gain some inspiration for working with multiple ports using groups too.
https://github.com/clientIO/joint/discussions/1761

Depending on your situation, you might be able to take advantage of some or our pre-defined port layouts too.
https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#layout.Port

**Reply by seekeramento** [reply]
Thanks. I think I will need time to check this before getting back to you.

**Reply by kumilingus** [reply]
Here is an example of the `absolute` port layout, where you can define each port position separately by specifying `x` and `y`.
https://jsfiddle.net/kumilingus/pztL4o1q/

**Reply by seekeramento** [reply]
Thanks. This will be helpful.
